keras/src/export/litert.py

```diff
@@ -0,0 +1,255 @@
+    generates a `.tflite` model file. For efficient inference on mobile and
+    embedded devices, it creates a single callable signature based on the
+    model's `call()` method.
+    """
+
+    def __init__(
+        self,
+        model,
+        input_signature=None,
+        **kwargs,
+    ):
+        """Initialize the LiteRT exporter.
+
+        Args:
+            model: The Keras model to export
+            input_signature: Input signature specification (e.g., TensorFlow
+                TensorSpec or list of TensorSpec)
+            **kwargs: Additional export parameters
+        """
+        self.model = model
+        self.input_signature = input_signature
+        self.kwargs = kwargs
+
+    def _infer_dict_input_signature(self):
+        """Infer input signature from a model with dict inputs.
+
+        This reads the actual shapes and dtypes from model._inputs_struct.
+
+        Returns:
+            dict or None: Dictionary mapping input names to InputSpec, or None
+        """
+        # Check _inputs_struct first (preserves dict structure)
+        if hasattr(self.model, "_inputs_struct") and isinstance(
+            self.model._inputs_struct, dict
+        ):
+            return {
+                name: make_input_spec(inp)
+                for name, inp in self.model._inputs_struct.items()
+            }
+
+        # Fall back to model.inputs if it's a dict
+        if hasattr(self.model, "inputs") and isinstance(
+            self.model.inputs, dict
+        ):
+            return {
+                name: make_input_spec(inp)
+                for name, inp in self.model.inputs.items()
+            }
+
+        return None
+
+    def export(self, filepath):
+        """Exports the Keras model to a TFLite file.
+
+        Args:
+            filepath: Output path for the exported model
+
+        Returns:
+            Path to exported model
+        """
+        # 1. Resolve / infer input signature
+        if self.input_signature is None:
+            # Try dict-specific inference first (for models with dict inputs)
+            dict_signature = self._infer_dict_input_signature()
+            if dict_signature is not None:
+                self.input_signature = dict_signature
+            else:
+                # Fall back to standard inference
+                self.input_signature = get_input_signature(self.model)
+
+        # 3. Handle dictionary inputs by creating an adapter
+        # Check if we have dict inputs that need adaptation
+        has_dict_inputs = isinstance(self.input_signature, dict)
+
+        if has_dict_inputs:
+            # Create adapter model that converts list to dict
+            adapted_model = self._create_dict_adapter(self.input_signature)
+
+            # Convert dict signature to list for TFLite conversion
+            # The adapter will handle the dict->list conversion
+            input_signature_list = list(self.input_signature.values())
```

Comment:
`input_signature_list = keras.tree.flatten(self.input_signature)`

This works for any nested structure, not just dicts.